src/main/java/com/recruitment/project/parking/parkingspaces/service/UserParkingVisitServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+        return userParkingVisitRepository.findAll();
+    }
+
+    @Override
+    public void startUserParkingVisit(String registrationNumber, boolean isDisabled, String brand, String model) {
+        UserVehicle userVehicle = userVehicleRepository.findByRegistration(registrationNumber);
+        if (userVehicle == null) {
+            userVehicle = new UserVehicle();
+            userVehicle.setRegistration(registrationNumber);
+            userVehicle.setBrand(brand);
+            userVehicle.setModel(model);
+            userVehicleRepository.save(userVehicle);
+        }
+        UserParkingVisit userParkingVisit = new UserParkingVisit();
+        userParkingVisit.setUserVehicle(userVehicle);
+        userParkingVisit.setDisabledUser(isDisabled);
+        userParkingVisit.setStartParking(LocalDateTime.now());
+        userParkingVisitRepository.save(userParkingVisit);
+    }
+
+    @Override
+    public void endUserParkingVisit(String registrationNumber) {
+        UserVehicle userVehicle = userVehicleRepository.findByRegistration(registrationNumber);
+        if (userVehicle != null) {
+            UserParkingVisit userParkingVisit = userParkingVisitRepository.findTopByUserVehicleOrderByStartParkingDesc(userVehicle);
+            userParkingVisit.setFinishParking(LocalDateTime.now());
+            Duration parkingVisitTime = Duration.between(userParkingVisit.getStartParking(), userParkingVisit.getFinishParking());
+            userParkingVisit.setCostVisit(parkingCost(parkingVisitTime.toMinutes(), userParkingVisit.isDisabledUser()));
+            userParkingVisitRepository.save(userParkingVisit);
+            logger.info(userParkingVisit.getStartParking().toString());
+        } else {
+            logger.info("Vehicle with registration number: " + registrationNumber + "did not started park");
+        }
+    }
+
+    @Override
+    public UserParkingVisit isStartingUserParkingVisit(String registrationNumber) {
+        UserVehicle userVehicle = userVehicleRepository.findByRegistration(registrationNumber);
+        if (userVehicle != null) {
+            UserParkingVisit userParkingVisit = userParkingVisitRepository.findTopByUserVehicleOrderByStartParkingDesc(userVehicle);
+            return userParkingVisit;
+        } else {
+            logger.info("Vehicle with registration number: " + registrationNumber + " did not started park");
+            return null;
+        }
+
+    }
+
+    @Override
+    public double getCostOneParkingVisit(String registrationNumber) {
+        UserVehicle userVehicle = userVehicleRepository.findByRegistration(registrationNumber);
+        if (userVehicle != null) {
+            UserParkingVisit userParkingVisit = userParkingVisitRepository.findTopByUserVehicleOrderByFinishParkingDesc(userVehicle);
+            return userParkingVisit.getCostVisit();
+        } else {
+            logger.info("Vehicle with registration number: " + registrationNumber + " did not started park");
+            return -1;
```

Comment:
Jak reszta systemu ma taką magiczną stałą obsłużyć?

Reply:
czy -1 oznacza, że to my mamy dopłacić, a nie kierowca? :)

Reply:
Miałem to dokończyć :)
Tutaj można by było przykłądowo stowrzyć własny wyjątek VehicleNotFound  i rzucić nim w tym serwisie, a obsłużyć w kontolerze.